courses/static_analysis_suite/150_lab_customization.rst

```diff
@@ -0,0 +1,241 @@
+
+===================
+Filtering Sources
+===================
+
+-----------------------
+Focusing Our Analysis
+-----------------------
+
+The Legacy module is raising the majority of error, and its code is using many dangerous constructs
+
+As a result we want to focus on the module that is not legacy.
+
+* In the locations tab, enter the name of :file:`cruise.adb` in the filter text box
+
+.. image:: codepeer/cruise_filter_in.png
+
+* This shows only messages for the specified file
+
+------------------
+Locations Filter
+------------------
+
+* Open the filter advanced menu by clicking on the magnifying glass icon in the filter bar.
+
+  .. image:: codepeer/cruise_filter_advanced_button.png
+    :width: 50%
+
+  .. image:: codepeer/cruise_filter_advanced_menu.png
+    :width: 50%
+
+  * Notice that you have several options there to filter with advanced messages criterias, as well as a search history.
+
+--------------------
+Filtering Messages
+--------------------
+
+* We want to sort the messages by importance, which is called sub-category in the tool.
+
+  * Click on the "hamburger" menu on the right hand side of the search bar.
+
+.. image:: codepeer/cruise_filter_configuration.png
+
+  * Select "Sort by subcategory"
+
+    * Notice that the messages are now displayed as high category first, then medium.
+
+------------------------
+Filtering Source Files
+------------------------
+
+* Filtering options on existing analysis are useful, but now we want to simply ignore the legacy files when performing the analysis.
+
+  * Open the project file from the :menu:`Project` tab
+
+    *  Right-click on :menu:`Cruise` |rightarrow| :menu:`Project` |rightarrow| :menu:`Edit Source File`.
+
+      .. image:: codepeer/radar_open_gpr.png
+
+* Ignore the :ada:`Cruise.Legacy` package in analysis
+
+  * In :toolname:`GPRBuild` you can set the :ada:`CodePeer.Excluded_Source_Files` attribute
+  * Use this as a template
+
+    .. code:: Ada
+
+        package CodePeer is
+            for Excluded_Source_Files use ("file_to_exclude.ads",
+                                           "second_file_to_exclude.adb");
+        end CodePeer;
+
+-------------------------------
+Analysis with Filtered Source
+-------------------------------
+
+* Run a new analysis and verify legacy files do not appear in the result anymore.
+
+  * In the Message History menu of the CodePeer Report, check only "Removed"
+  * Several warnings should be displayed
+  * Expand the :ada:`Cruise` project to see all files
+
+* The result should show that those removed warnings belonged to :file:`cruise-legacy.adb`
+
+.. image:: codepeer/cruise_exclude_legacy.png
+
+=================
+Change Settings
+=================
+
+-------------------------------
+Settings Via The Project File
+-------------------------------
+
+* We want to set the default value of the CodePeer parameters in the GPR file.
+
+  * The :toolname:`GPRBuild` attribute to set the switches is :ada:`CodePeer.Switches`
```

Comment:
I would recommend providing example code here (as was done for the excluded files) to reduce confusion